src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Extensibility/PluginsManagerResourceReflector.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Performance.SDK.Runtime;
+using Microsoft.Performance.SDK.Runtime.Discovery;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Extensibility
+{
+    /// <summary>
+    ///     This class registers to with a provider to receive Types that might be
+    ///     <see cref="Plugins.Core.Extensibility.IPluginsManagerResource"/>s.
+    ///     The types are evaluated and stored as a <see cref="PluginsManagerResourceReference"/> when applicable.
+    /// </summary>
+    public sealed class PluginsManagerResourceReflector
+        : IExtensionTypeObserver
+    {
+        private readonly Dictionary<Type, PluginsManagerResourceReference> loadedResources;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="PluginsManagerResourceReflector"/>
+        ///     class, registering to receive updates from the given <see cref="IExtensionTypeProvider"/>
+        /// </summary>
+        /// <param name="extensionTypeProvider">
+        ///     The object doing extension discovery.
+        /// </param>
+        public PluginsManagerResourceReflector(IExtensionTypeProvider extensionTypeProvider)
+        {
+            this.loadedResources = new Dictionary<Type, PluginsManagerResourceReference>();
+            extensionTypeProvider.RegisterTypeConsumer(this);
+        }
+
+        /// <inheritdoc />
+        public void ProcessType(Type type, string sourceName)
+        {
+            if (PluginsManagerResourceReference.TryCreateReference(type, out PluginsManagerResourceReference reference))
+            {
+                try
+                {
+                    this.loadedResources.Add(type, reference);
+                }
+                catch { }
```

Comment:
should we catch and log/return an error?

Reply:
The reflector is disabled right now. Added a TODO here so that we handle errors once it's re-enabled.